modules/crypto.js

```diff
@@ -0,0 +1,285 @@
+// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
+// Copyright Joyent, Inc. and Node.js contributors. All rights reserved. MIT license.
+
+import { ERR_CRYPTO_FIPS_FORCED, ERR_CRYPTO_TIMING_SAFE_EQUAL_LENGTH, ERR_INVALID_ARG_TYPE } from "./internal/errors";
+import { crypto as constants } from "./internal_binding/constants";
+import { getOptionValue } from "./internal/options";
+import { isAnyArrayBuffer, isArrayBufferView } from "./internal/util/types";
+import {
+  timing_safe_equal,
+} from "_node:crypto";
+function timingSafeEqual(a, b) {
+  if (!isAnyArrayBuffer(a) && !isArrayBufferView(a)) {
+    throw new ERR_INVALID_ARG_TYPE("buf1", ["ArrayBuffer", "Buffer", "TypedArray", "DataView"], a);
+  }
+  if (!isAnyArrayBuffer(b) && !isArrayBufferView(b)) {
+    throw new ERR_INVALID_ARG_TYPE("buf2", ["ArrayBuffer", "Buffer", "TypedArray", "DataView"], b);
+  }
+  if (a.byteLength != b.byteLength) {
+    throw new ERR_CRYPTO_TIMING_SAFE_EQUAL_LENGTH();
+  }
+  return timing_safe_equal(a.buffer, b.buffer);
+}
+
+import {
+  checkPrime,
+  checkPrimeSync,
+  generatePrime,
+  generatePrimeSync,
+  randomBytes,
+  randomFill,
+  randomFillSync,
+  randomInt,
+  randomUUID,
+} from "./internal/crypto/random";
+import { pbkdf2, pbkdf2Sync } from "./internal/crypto/pbkdf2";
+import { scrypt, scryptSync } from "./internal/crypto/scrypt";
+import { hkdf, hkdfSync } from "./internal/crypto/hkdf";
+/*import {
+  generateKey,
+  generateKeyPair,
+  generateKeyPairSync,
+  generateKeySync,
+} from "./internal/crypto/keygen";
+import {
+  createPrivateKey,
+  createPublicKey,
+  createSecretKey,
+  KeyObject,
+} from "./internal/crypto/keys";
+import {
+  DiffieHellman,
+  diffieHellman,
+  DiffieHellmanGroup,
+  ECDH,
+} from "./internal/crypto/diffiehellman";*/

```

Comment:
Why comment this code when you have `diffiehellman.js`?

Reply:
It hasn't implemented for wasi and tested. It's just copied from nodejs now.